Ervin Lance Zamora - PDF Quiz Generator

CnRXoMoX opened the pull request:
Some highlights of my implementation:
- In-browser PDF parsing with type-safe handling using pdf.js
- AI quiz generation powered by a Groq-hosted LLaMA 4 model, with robust JSON extraction and error handling
- Clean and accessible React UI for taking quizzes, including question navigation and scoring
- Strong TypeScript and ESLint enforcement with careful error handling and type narrowing to ensure reliability
- Custom toast notification hooks for seamless user feedback

CnRXoMoX committed:
Initial commit: PDF Quiz Generator with PDF parsing, AI quiz generation, and quiz UI + TS fixes

This initial commit implements the core functionality of the PDF Quiz Generator project, including:

- PDF parsing in-browser using pdf.js with type-safe extraction of text content
- Integration with Groq-hosted LLaMA 4 model for AI-powered quiz generation from PDF text
- React-based quiz UI with question navigation, scoring, and reset functionality
- Robust form handling with React Hook Form and Zod validation
- Toast notifications via Sonner with custom useMessage hook
- Comprehensive TypeScript improvements:
  - Replaced explicit any types with proper type guards and defined interfaces
  - Added custom module declarations for pdfjs-dist to resolve missing types
  - Fixed ESLint errors related to hook usage and type safety in catch blocks
  - Added null checks and optional chaining to prevent runtime errors
- Detailed inline comments for better code readability and maintainability
- README.md outlining setup instructions, libraries used, design decisions, limitations, and future improvements

CnRXoMoX committed:
Update README.md

Bowis commented:
<img width="954" height="446" alt="image" src="https://github.com/user-attachments/assets/ff7ae303-c47d-4dbc-8f72-153abe14e658" />
Getting this error when trying to run this

CnRXoMoX committed:
Fix PDF.js import for browser compatibility in parsePdf

Replaced 'pdfjs-dist/build/pdf' with 'pdfjs-dist/webpack' to avoid server-side canvas dependency errors during Next.js build.

- Ensured compatibility with client-side parsing only

Bowis commented:
> The pdfjs import is running a node version of it, at my end it wasn't an issue, even tho we use client only the problem lies at the pdfjs-dist/build/pdf. Replaced the import to pdfjs-dist/webpack to resolve this issue and it's safe for browser use and it won't try to require canvas. Gonna commit the fix now.

Hmm, still getting the same error! What node version are you using?

CnRXoMoX commented:
> > The pdfjs import is running a node version of it, at my end it wasn't an issue, even tho we use client only the problem lies at the pdfjs-dist/build/pdf. Replaced the import to pdfjs-dist/webpack to resolve this issue and it's safe for browser use and it won't try to require canvas. Gonna commit the fix now.
> 
> Hmm, still getting the same error! What node version are you using?

I am using v20.10.0
